cmake/modules/OpenCL.cmake

```diff
@@ -49,12 +49,17 @@ else()
 endif(USE_AOCL)
 
 if(USE_OPENCL)
-  if (NOT OpenCL_FOUND)
-    find_package(OpenCL REQUIRED)
-  endif()
-  message(STATUS "Build with OpenCL support")
   tvm_file_glob(GLOB RUNTIME_OPENCL_SRCS src/runtime/opencl/*.cc)
-  list(APPEND TVM_RUNTIME_LINKER_LIBS ${OpenCL_LIBRARIES})
+  if (OpenCL_FOUND)
+    message(STATUS "Build with OpenCL support")
```

Comment:
What are the cases when we find opencl and link vs it? Do we assume this wrapper not cover all cases and we will need to link vs original opencl?

Reply:
This wrapper can cover all cases. I kept this search only for case which I have described in the comment to your question about OpenCL SDK. Actually, probably we can remove search of OpenCL and use wrapper in all cases. I didn't do it only because the user might want to specify a specific version of OpenCL or OpenCL SDK and I think we should give him/her this opportunity.

Reply:
This can happen only if this specific version of OpenCL has functions which are not in the OpenCL standard. I hardly can imagine that tvm OpenCL runtime can depend on such functions. All specific can be in OpenCL extensions which can be a part of kernels, but not host API. Let's completely remove flow with static linking and depending on OpenCL SDK

Reply:
Eventually we decided to leave static OpenCL linking but change conditions when it is selected and when wrapper should be selected. There might be cases when dlopen is prohibited due to sandbox limitations and only libraries statically selected by loader can be opened.
Modificatoin of USE_OPENCL logic: 
if USE_OPENCL has ON, always use wrapper and dynamic loading of libOpenCL.so in runtime
if USE_OPENCL points to openCL SDK, then use this sdk and static linking without wrapper